src/TMap.cpp

```diff
@@ -1521,7 +1527,7 @@ bool TMap::restore(QString location, bool downloadIfNotFound)
         if (!foundValidFile) {
             canRestore = false;
         }
-    } else {
+    } else if (!noMapFile) {
```

Comment:
Isn't this the same condition as the `if` clause now? According to DeMorgan: with `noMapFile == (entries.empty && location.empty)` follows `!(entries,empty && location.empty) == (!entries.empty || !location.empty)`

Reply:
To reiterate and explain: I think the if in line 1491 and the if of the else belonging to that if (line 1530) have the same condition. Because noMapFile is only true if both entries and location are empty. Since both variables don't change between setting the variable and the if clause, you can expand the variable again and use DeMorgan as I showed above.

Reply:
Okay - reworked it - if it doesn't fit feel free to rework it

Reply:
@keneanung mind double-checking?